modules/rpc/yarpc.go

```diff
@@ -62,13 +69,30 @@ var (
 
 type registerServiceFunc func(module *YARPCModule)
 
+type transports struct {
+	inbounds []transport.Inbound
+}
+
 type yarpcConfig struct {
 	modules.ModuleConfig
-	Bind                    string
-	AdvertiseName           string `yaml:"advertiseName"`
+
 	inboundMiddleware       []middleware.UnaryInbound
 	onewayInboundMiddleware []middleware.OnewayInbound
-	inbounds                []transport.Inbound
+
+	transports transports
+	Inbounds   []Inbound
+}
+
+// Inbound is a union that configures how to configure a single inbound.
+type Inbound struct {
+	TChannel *Address
+	HTTP     *Address
+}
+
+// Address is a struct that have a required port for tchannel/http transports.
+// TODO(alsam) make it optional
+type Address struct {
```

Comment:
can this not be just a type `Port`? or is there anything else going to be in Address?
```
type Port int
type Inbound struct {
	TChannel *Port
	HTTP     *Port
}
```